obspy/io/sac/sactrace.py

```diff
@@ -1105,12 +1102,15 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
 
         .. rubric:: Example
 
-        >>> sac = SACTrace.read(filename, headonly=True) # doctest: +SKIP
-        >>> try: # doctest: +SKIP
-                sac.validate('data_hdrs') # doctest: +SKIP
-            except SacInvalidContentError: # doctest: +SKIP
-                sac = SACTrace.read(filename, headonly=False) # doctest: +SKIP
-                sac.validate('data_hdrs') # doctest: +SKIP
+        >>> from obspy.core.util import get_example_file
+        >>> from obspy.io.sac.util import SacInvalidContentError
+        >>> file_ = get_example_file("test.sac")
+        >>> sac = SACTrace.read(file_, headonly=True)
+        >>> try:
+        ...     sac.validate('data_hdrs')
+        ... except (ValueError, SacInvalidContentError):
+        ...     sac = SACTrace.read(file_, headonly=False)
+        ...     sac.validate('data_hdrs')
```

Comment:
Hmmm, I chose to raise `ValueError` for the `data_hdrs` check.  Your change looks correct.  Thanks!  Maybe I should also raise a `SacInvalidContentError` with an appropriate message instead.

BTW, I actually get a `SacInvalidContentError` when I validate with the `headonly=False` read, because as it turns out the mean of the data in `test.sac` is not what's in the what's in the file header.  This explains the [other error](https://ci.appveyor.com/project/obspy/obspy/build/1.0.2450-better_SACTrace_docs#L830).